lab_1/main.py

```diff
@@ -86,34 +153,98 @@ def compare_profiles_advanced(unknown_profile: dict, profile_to_compare: dict, t
     :return: a dictionary with 7 keys – name, score, common, sorted_common, max_length_word,
     min_length_word, average_token_length
     """
-    pass
-
-
-def detect_language_advanced(unknown_profile: dict, profiles: list, languages: list, top_n: int) -> str or None:
+    if isinstance(unknown_profile, dict) \
+            and isinstance(profile_to_compare, dict) \
+            and isinstance(top_n, int):
+        common = []
+        top_n_words_unknown_profile = get_top_n_words(unknown_profile['freq'], top_n)
+        top_n_words_known_profile = get_top_n_words(profile_to_compare['freq'], top_n)
+        for i in top_n_words_known_profile:
+            if i in top_n_words_unknown_profile:
+                common.append(i)
+        sorted_common = sorted(common)
+        score = (len(common) / len(top_n_words_known_profile))
+        profile_to_compare_keys = list(profile_to_compare['freq'].keys())
+        sorted_keys = sorted(profile_to_compare_keys, key=len)
```

Comment:
can you explain what happens here? In writing

Reply:
first of all, we get **top n words of unknown and known language** using function get_top_n_words. then we use these variables to get list of **common words** in cycle. then we divide number of common words by nuber of top n words of unknown language and in the end we get **score** (proportion of  intersecting words). then the length of the **shortest word and the longest** is counted here(profile_to_compare_keys = list(profile_to_compare['freq'].keys())- pull words out of the frequency dictionary, sort them by length and by indices we take the first and last word)
and the last variable **average_token_length**(put all words of the frequency dictionary in one word, take its length and divide it by number of these words). and then all these variables is used as **values in dictionary of the full profile of unknown language**